### PR TITLE
feat: Swaps consume bip44 default pairs from LD

### PR DESCRIPTION
**Pull request by infiniteflower**
## **Description**

This PR consumes BIP44 default pairs from LaunchDarkly. If remote flags cannot be reached, we default back to local chain specific default pairs.

1. If you enter Swaps not from the Asset Details page with "All Popular Networks" filter, we use the BIP44 default pairs from LD
2. If you enter Swaps from the Asset Details page, we use the per chain default pairs
3. If you change the network filter on the Home screen, we respect that and use the per chain default pairs

## **Changelog**

CHANGELOG entry: null

## **Related issues**

Resolves: https://consensyssoftware.atlassian.net/browse/SWAPS-3033

## **Manual testing steps**

```gherkin
Feature: LD BIP44 default pairs

  Scenario: user enters swaps
    Given user has entered swaps

    When user enters swaps
    Then Swaps should populate with the BIP44 default pairs from LD
```

### **After**

https://github.com/user-attachments/assets/5611d847-94cc-4bd7-8cb8-b6d8c6dec091

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> Swaps now preselect tokens from LD BIP44 default pairs and bridge tx submission passes the wallet address; bridge controllers upgraded.
> 
> - **Swaps/Bridge UI**:
>   - Use remote feature flags to derive defaults via new `selectBip44DefaultPair`; fallback maintained.
>   - `useInitialSourceToken`/`useInitialDestToken` set initial tokens from BIP44 pairs when none provided.
>   - Add `Bip44TokensForDefaultPairs` and extend `DefaultSwapDestTokens` (names/images tweaks).
> - **Redux (bridge slice)**:
>   - New selector `selectBip44DefaultPair` (parses CAIP, maps to `Bip44TokensForDefaultPairs`).
>   - Tests added for multiple namespaces and edge cases.
> - **Bridge TX submission**:
>   - Update calls to `BridgeStatusController.submitTx(from, quote, stxEnabled)`; fetch `from` via `selectSourceWalletAddress` and validate.
>   - Adjust pay hook and related tests to pass `from`.
> - **Messengers**:
>   - Replace allowed action `AccountsController:getSelectedMultichainAccount` with `AccountsController:getAccountByAddress` in bridge messengers.
> - **Dependencies**:
>   - Bump `@metamask/bridge-controller` and `@metamask/bridge-status-controller` to `^47.2.0`; add Yarn patch and small optional-chaining fix in dependency.
> - **Tests/Mocks**:
>   - Update mocks for `bip44DefaultPairs` and token metadata; adjust expectations for images and Solana-native handling.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 66153a44e103cada40d198aa79ad1bfa5993e764. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: app/core/redux/slices/bridge/index.ts [hunk 1]
```diff
@@ -496,6 +502,35 @@ export const selectNoFeeAssets = createSelector(
...
+      Bip44TokensForDefaultPairs[sourceAssetId as CaipAssetType];
+    const destAsset = Bip44TokensForDefaultPairs[destAssetId as CaipAssetType];
+
+    if (!sourceAsset || !destAsset) {
+      return undefined;
+    }
+
+    return { sourceAsset, destAsset };
+  },
+);
```
[the remark this answers is not in the document]
Comment (infiniteflower) [reply app/core/redux/slices/bridge/index.ts:532]: You are wrong

File: app/components/UI/Bridge/hooks/useInitialSourceToken/index.ts [hunk 2]
```diff
@@ -69,14 +75,21 @@ export const useInitialSourceToken = (
...
   const chainId = isEvmNetworkSelected
     ? selectedEvmChainId
     : selectedNonEvmNetworkChainId;
 
   useEffect(() => {
-    // Will default to the native token of the current chain if no token is provided
+    // If no initial source token is provided,
+    // set the source token to the bip44 default pair (preferred) or the native token of the current chain
     if (!initialSourceToken && !sourceToken) {
+      if (bip44DefaultPair) {
```
Comment (SteP-n-s) [inline app/components/UI/Bridge/hooks/useInitialSourceToken/index.ts:88]: Do we need to detect if it's swap like we do in the `useInitialDestToken` hook?

`if (isSwap && bip44DefaultPair && !destToken) {...}`

Comment (infiniteflower) [reply app/components/UI/Bridge/hooks/useInitialSourceToken/index.ts:88]: No because we would always be setting the source, only in dest is it optional.

E.g. in Bridge we didn't set the dest token because we don't know what network they want to go to. But even in Bridge we always set the source regardless because we knew what network they were on.

**Review by SteP-n-s — APPROVED** [review]

**Review by smilingkylan — APPROVED** [review]
LGTM ✅
